cep18/src/entry_points.rs

```diff
@@ -0,0 +1,222 @@
+    )
+}
+
+/// Returns the `decrease_allowance` entry point.
+pub fn decrease_allowance() -> EntryPoint {
+    EntryPoint::new(
+        String::from(DECREASE_ALLOWANCE_ENTRY_POINT_NAME),
+        vec![
+            Parameter::new(SPENDER, Key::cl_type()),
+            Parameter::new(AMOUNT, U256::cl_type()),
+        ],
+        CLType::Unit,
+        EntryPointAccess::Public,
+        EntryPointType::Contract,
+    )
+}
+
+/// Returns the `transfer` entry point.
+pub fn transfer() -> EntryPoint {
+    EntryPoint::new(
+        String::from(TRANSFER_ENTRY_POINT_NAME),
+        vec![
+            Parameter::new(RECIPIENT, Key::cl_type()),
+            Parameter::new(AMOUNT, U256::cl_type()),
+        ],
+        CLType::Unit,
+        EntryPointAccess::Public,
+        EntryPointType::Contract,
+    )
+}
+
+/// Returns the `balance_of` entry point.
+pub fn balance_of() -> EntryPoint {
+    EntryPoint::new(
+        String::from(BALANCE_OF_ENTRY_POINT_NAME),
+        vec![Parameter::new(ADDRESS, Key::cl_type())],
+        U256::cl_type(),
+        EntryPointAccess::Public,
+        EntryPointType::Contract,
+    )
+}
+
+/// Returns the `total_supply` entry point.
+pub fn total_supply() -> EntryPoint {
+    EntryPoint::new(
+        String::from(TOTAL_SUPPLY_ENTRY_POINT_NAME),
+        Vec::new(),
+        U256::cl_type(),
+        EntryPointAccess::Public,
+        EntryPointType::Contract,
+    )
+}
+
+/// Returns the `decimals` entry point.
+pub fn decimals() -> EntryPoint {
+    EntryPoint::new(
+        String::from(DECIMALS_ENTRY_POINT_NAME),
+        Vec::new(),
+        u8::cl_type(),
+        EntryPointAccess::Public,
+        EntryPointType::Contract,
+    )
+}
+
+/// Returns the `burn` entry point.
+pub fn burn() -> EntryPoint {
+    EntryPoint::new(
+        String::from(BURN_ENTRY_POINT_NAME),
+        vec![Parameter::new(AMOUNT, U256::cl_type())],
+        CLType::Unit,
+        EntryPointAccess::Public,
+        EntryPointType::Contract,
+    )
+}
+
+/// Returns the `mint` entry point.
+pub fn mint() -> EntryPoint {
+    EntryPoint::new(
+        String::from(MINT_ENTRY_POINT_NAME),
+        vec![Parameter::new(AMOUNT, U256::cl_type())],
+        CLType::Unit,
+        EntryPointAccess::Public,
+        EntryPointType::Contract,
+    )
+}
+
+/// Returns the `change_security` entry point.
+pub fn change_security() -> EntryPoint {
+    EntryPoint::new(
+        String::from(CHANGE_SECURITY_ENTRY_POINT_NAME),
+        vec![],
+        CLType::Unit,
+        EntryPointAccess::Public,
+        EntryPointType::Contract,
+    )
+}
+
+/// Returns the `init` entry point.
+pub fn init() -> EntryPoint {
+    EntryPoint::new(
+        String::from(INIT_ENTRY_POINT_NAME),
+        Vec::new(),
+        CLType::Unit,
+        EntryPointAccess::Public,
+        EntryPointType::Contract,
+    )
+}
+
+/// Returns the default set of CEP18 token entry points.
+pub fn generate_entry_points(enable_mint_burn: bool) -> EntryPoints {
```

Comment:
Lets change this behavior to ensure we preserve the modality to the named keys, and within the entrypoints check the modality and early exit if mint and burn are disabled